### PR TITLE
Spruce up docs on pull request builds

### PR DESCRIPTION
**Pull request by agjohnson**
I keep helping users with the support request around webhooks missing
the pull request event, so wanted to clarify a bit here. I also took the
opportunity to add some clearer language to the rest of the page, and
resturcture a few things to improve the information flow.

I also opened #9176 to deal with this in a programmatic way, but figured
we could at least communicate the issue better for now.

### CODE REVIEW

File: docs/user/pull-requests.rst [hunk 1]
```diff
@@ -1,37 +1,49 @@
...
 Build on pull request events
-    We create and build a new version when a pull request is open,
-    and when a new commit has been pushed.
+    We create and build a new version when a pull request is opened,
+    and rebuild the version whenever a new commit is pushed.
 
 Build status report
-    When a build is triggered, a build pending notification is sent with a link to the build log.
-    When the build finishes we send a success notification with the link to the preview or a failure notification with a link to the build log.
+    Your project's pull request build status will show as as one of your pull
```
Comment (stsewd) [inline docs/user/pull-requests.rst:17]:
```suggestion
    Your project's pull request build status will show as one of your pull
```

**Review by stsewd — APPROVED** [review]

File: docs/user/pull-requests.rst [hunk 2]
```diff
@@ -40,43 +52,60 @@ Privacy levels
 
    Privacy levels are only supported on :doc:`/commercial/index`.
 
-By default all docs built from pull requests are private.
+By default, all docs built from pull requests are private.
 To change their privacy level:
 
 #. Go to your project dashboard
-#. Go to :guilabel:`Admin > Advanced settings`
+#. Go to :guilabel:`Admin`, then :guilabel:`Advanced settings`
```
Comment (humitos) [inline docs/user/pull-requests.rst:59]: I think we agreed on using the `>` syntax and we have been using it a lot on multiple pages.

**Review by humitos — APPROVED** [review]

File: docs/user/pull-requests.rst (hunk 2 above)
Comment (agjohnson) [reply docs/user/pull-requests.rst:59]: I might have missed this conversation, but would say that `>` is acceptable, but verbose language is preferable -- neither are wrong though. Microsoft has the clearest style guide here:
https://docs.microsoft.com/en-us/style-guide/procedures-instructions/describing-interactions-with-ui

Using `>` is acceptable, but preference is on using verbose language. The issues with using `>` are accessibility concerns. The `>` may not render correctly for RTL site translations, and likely sounds awful to screen readers.

Here's the page in Hebrew for example, note Google does invert the direction of `>` in some cases:

image

So, either way, this should not be ```:guilabel:`Admin > Advanced settings` ``` as these are two separate GUI elements. But also, RTL site translation breaks in this case.

Comment (humitos) [reply docs/user/pull-requests.rst:59]: Good analysis! 💯 

> So, either way, this should not be `` :guilabel:`Admin > Advanced settings` `` as these are two separate GUI elements.

Definitely, not this option 😄

**Pull request merged**